google/cloud/aiplatform/metadata/experiment_resources.py

```diff
@@ -0,0 +1,717 @@
+            rows.append(row_dict)
+
+        df = pd.DataFrame(rows)
+
+        column_name_sort_map = {
+            "experiment_name": -1,
+            "run_name": 1,
+            "run_type": 2,
+            "state": 3,
+        }
+
+        def column_sort_key(key: str) -> int:
+            """Helper method to reorder columns."""
+            order = column_name_sort_map.get(key)
+            if order:
+                return order
+            elif key.startswith("param"):
+                return 5
+            elif key.startswith("metric"):
+                return 6
+            else:
+                return 7
+
+        columns = df.columns
+        columns = sorted(columns, key=column_sort_key)
+        df = df.reindex(columns, axis=1)
+
+        return df
+
+    def _lookup_backing_tensorboard(self) -> Optional[tensorboard_resource.Tensorboard]:
+        """Returns backing tensorboard if one is set.
+
+        Returns:
+            Tensorboard resource if one exists.
+        """
+        tensorboard_resource_name = self._metadata_context.metadata.get(
+            constants._BACKING_TENSORBOARD_RESOURCE_KEY
+        )
+
+        if not tensorboard_resource_name:
+            with _SetLoggerLevel(resource):
+                self._metadata_context.sync_resource()
+            tensorboard_resource_name = self._metadata_context.metadata.get(
+                constants._BACKING_TENSORBOARD_RESOURCE_KEY
+            )
+
+        if tensorboard_resource_name:
+            return tensorboard_resource.Tensorboard(
+                tensorboard_resource_name,
+                credentials=self._metadata_context.credentials,
+            )
+
+    def get_backing_tensorboard_resource(
+        self,
+    ) -> Optional[tensorboard_resource.Tensorboard]:
+        """Get the backing tensorboard for this experiment in one exists.
+
+        ```
+        my_experiment = aiplatform.Experiment('my-experiment')
+        tb = my_experiment.get_backing_tensorboard_resource()
+        ```
+
+        Returns:
+            Backing Tensorboard resource for this experiment if one exists.
+        """
+        return self._lookup_backing_tensorboard()
+
+    def assign_backing_tensorboard(
+        self, tensorboard: Union[tensorboard_resource.Tensorboard, str]
+    ):
+        """Assigns tensorboard as backing tensorboard to support time series metrics logging.
+
+        ```
+        tb = aiplatform.Tensorboard('tensorboard-resource-id')
+        my_experiment = aiplatform.Experiment('my-experiment')
+        my_experiment.assign_backing_tensorboard(tb)
+        ```
+
+        Args:
+            tensorboard (Union[aiplatform.Tensorboard, str]):
+                Required. Tensorboard resource or resource name to associate to this experiment.
+
+        Raises:
+            ValueError: If this experiment already has a previously set backing tensorboard resource.
+            ValueError: If Tensorboard is not in same project and location as this experiment.
```

Comment:
Should this also raise if the provided Tensorboard is not valid / doesn't exist? Or will the service raise an error in that case?

Reply:
The service will raise an exception in that case.